src/executor/stack/executor.rs

```diff
@@ -957,38 +964,54 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet>
 		let state = self.state_mut();
 		let mut warm_authority: Vec<H160> = Vec::with_capacity(authorization_list.len());
 		for authority in authorization_list {
-			// If EIP-7703 Spec validation steps 1 or 2 return false.
+			// If EIP-7702 Spec validation steps 1, 3 return false.
 			if !authority.is_valid {
 				continue;
 			}
-			// 3. Add authority to accessed_addresses (as defined in EIP-2929)
+
+			// 2. Verify the `nonce` is less than `2**64 - 1`.
+			if U256::from(authority.nonce) >= U64_MAX {
+				continue;
+			}
+
+			// 4. Add authority to accessed_addresses (as defined in EIP-2929)
 			warm_authority.push(authority.authority);
-			// 4. Verify the code of authority is either empty or already delegated.
+			// 5. Verify the code of authority is either empty or already delegated.
 			let authority_code = state.code(authority.authority);
 			if !authority_code.is_empty() && !Authorization::is_delegated(&authority_code) {
 				continue;
 			}
 
-			// 5. Verify the nonce of authority is equal to nonce.
+			// 6. Verify the nonce of authority is equal to nonce.
 			if state.basic(authority.authority).nonce != U256::from(authority.nonce) {
 				continue;
 			}
 
-			// 6. Add PER_EMPTY_ACCOUNT_COST - PER_AUTH_BASE_COST gas to the global refund counter if authority exists in the trie.
+			// 7. Add PER_EMPTY_ACCOUNT_COST - PER_AUTH_BASE_COST gas to the global refund counter if authority exists in the trie.
 			if !state.is_empty(authority.authority) {
 				refunded_accounts += 1;
 			}
-			// 7. Set the code of authority to be `0xef0100 || address`. This is a delegation designation.
-			state.set_code(authority.authority, authority.delegation_code());
-			// 8. Increase the nonce of authority by one.
+			// 8. Set the code of authority to be `0xef0100 || address`. This is a delegation designation.
+			// * As a special case, if address is 0x0000000000000000000000000000000000000000 do not write the designation.
+			//   Clear the account’s code.
+			let mut delegation_clearing = false;
+			if authority.address.is_zero() {
+				delegation_clearing = true;
+				state.set_code(authority.authority, Vec::new());
+			} else {
+				state.set_code(authority.authority, authority.delegation_code());
+			}
```

Comment:
I guess rust-idiomatic here would be:
```rust
let delegation_clearing = if authority.address.is_zero() {
    state.set_code(authority.authority, Vec::new());
    true
} else {
	state.set_code(authority.authority, authority.delegation_code());
    false
};
```

Reply:
💯, thanks. It will be managed in the next PR that is already in progress.